local/compose/exec.go

```diff
@@ -42,10 +42,10 @@ func (s *composeService) Exec(ctx context.Context, project *types.Project, opts
 		),
 	})
 	if err != nil {
-		return err
+		return 0, err
 	}
 	if len(containers) < 1 {
-		return fmt.Errorf("container %s not running", getContainerName(project.Name, service, opts.Index))
+		return 0, fmt.Errorf("container %s not running", getContainerName(project.Name, service, opts.Index))
```

Comment:
do we really want this to be 0? 🤔

Reply:
with another language I would have used `Either(ExitCode, Error)` but with golang we have to return both values for "everything went fine" and "oups" scenario, as a tuple.
and we can't use `nil` here until we make the signature to use `*int` which is not really better

Reply:
as an additionl note, this is basically a copy/paste/adapt signature from https://github.com/docker/cli/blob/a32cd16160f1b41c1c4ae7bee4dac929d1484e59/cli/command/container/exec.go#L116
I'd like we can just rely on docker/cli for such interactions and don't have to duplicate code, maybe we can evolve in this direction making required changes to docker/cli once Compose 2.0 is a thing